src/network/networktool.cpp

```diff
@@ -0,0 +1,178 @@
+#include "networkmanager.h"
+#include "networktool.h"
+#include "notificationmanager.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QRegularExpression>
+#include <QTimer>
+#include <QUrl>
+
+PING_LOGGING_CATEGORY(NETWORKTOOL, "ping.networktool")
+
+void NetworkTool::checkInterfaceUpdate()
+{
+    static QString gitUserRepo;
+    if(gitUserRepo.isEmpty()) {
+        //https://github.com/user/repo.git
+        const static QRegularExpression regex("github.com/(.*).git");
+        QRegularExpressionMatch regexMatch = regex.match(QStringLiteral(GIT_URL));
+        // TODO: Check regex output
+        if (regexMatch.hasMatch()) {
+            gitUserRepo = regexMatch.capturedTexts()[1];
+        } else {
+            qCWarning(NETWORKTOOL) <<
+                                   "Fail to get github user and repository! It'll not be possible to check for updates";
+            return;
+        }
+    }
+
+    const QUrl url{QStringLiteral("https://api.github.com/repos/%1/releases").arg(gitUserRepo)};
+    NetworkManager::self()->requestJson(
+        url,
+        self()->checkNewVersionInGitHubPayload
+    );
+}
+
+void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
+{
+    const static QString projectTag = QStringLiteral(GIT_TAG);
+    static QRegularExpression releaseTagRegex;
+    // Check if version start with 't' from test version
+    // For everything we can presume normal release
+    if(projectTag.contains('t') == 0) {
+        releaseTagRegex.setPattern(QStringLiteral("(t)\\d+\\.\\d+"));
+        qCDebug(NETWORKTOOL) << "Running test version:" << projectTag;
+    } else {
+        releaseTagRegex.setPattern(QStringLiteral("(v)\\d+\\.\\d+"));
+        qCDebug(NETWORKTOOL) << "Running release version:" << projectTag;
+    }
+
+    // Check if this actual version is a real release
+    // Alarm people to download the correction version !
+    QRegularExpressionMatch regexMatch = releaseTagRegex.match(projectTag);
+    float actualVersion = regexMatch.hasMatch() ? projectTag.right(1).toFloat() : -1.0;
+
+    auto versionsAvailable = jsonDocument.array();
+    if(versionsAvailable.isEmpty()) {
+        qCWarning(NETWORKTOOL) << "GitHub json it's not an array!";
+        qCDebug(NETWORKTOOL) << jsonDocument;
+        return;
+    }
+
+    struct {
+        float version;
+        QString versionString;
+        QJsonValue json;
+    } lastReleaseAvailable;
+    lastReleaseAvailable.version = actualVersion;
+
+    // Check payload
+    for(const QJsonValue& versionPayload : versionsAvailable) {
+        auto versionString = versionPayload[QStringLiteral("tag_name")].toString();
+        qCDebug(NETWORKTOOL) << "Testing version:" << versionString;
+
+        regexMatch = releaseTagRegex.match(versionString);
+        if(!regexMatch.hasMatch()) {
+            continue;
+        }
+        qCDebug(NETWORKTOOL) << "Possible new version.";
+
+        auto version = versionString.right(1).toFloat();
```

Comment:
I think we need to talk about our versioning standards organization-wide. I like the simplicity of being able to simply parse a float, but I have a preference for Major.Minor.Patch semantic versioning.
https://semver.org/

Reply:
In my mind, it depends of what kind of release cycle we want to use in the project.
For APIs, semver makes sense. For HUGE projects cmsver. And for simple projects calver or this one that I'm using, that is just new_versions.patches;

Reply:
@Williangalvani what is your thought?

Reply:
@patrickelectric I agree generally, but I think there is benefit to consistency in our organization or something we can add to our guidelines. Something that our users and engineers (current and future) can get used to. I agree that x.x.x may be more than we need here, but it's not _that_ excessive in my opinion and will be able to cover every project we do.

We don't need to reach decision before merge, but before we cut first stable.

ping @Williangalvani.

Reply:
I agree with @jaxxzer that consistency is important.
I understand that it is not required in this project right now, and a little more annoying to code with, but I vote for semver.
Or maybe we could have x.x standard for smaller projects?
Also, is your code assuming the version is only 1 digit?